user/api.go

```diff
@@ -176,197 +206,213 @@ func (a *Api) CreateChildUser(res http.ResponseWriter, req *http.Request) {
-						sendModelAsResWithStatus(res, status.NewStatus(http.StatusConflict, STATUS_USR_ALREADY_EXISTS), http.StatusConflict)
-						return
-					}
+				if len(updatesToApply.Updates.Emails) > 0 {
+					userToUpdate.Emails = updatesToApply.Updates.Emails
+					dupCheck.Emails = userToUpdate.Emails
 				}
-				//Rehash the pw if needed
-				if updatesToApply.Updates.Pw != "" {
-
-					if err := userToUpdate.HashPassword(updatesToApply.Updates.Pw, a.ApiConfig.Salt); err != nil {
-						log.Printf(USER_API_PREFIX+"UpdateUser %s err[%s]", STATUS_ERR_UPDATING_USR, err.Error())
-						sendModelAsResWithStatus(res, status.NewStatus(http.StatusInternalServerError, STATUS_ERR_UPDATING_USR), http.StatusInternalServerError)
-						return
-					}
+				//check if unique
+				if results, err := a.Store.FindUsers(dupCheck); err != nil {
+					a.logger.Println(http.StatusInternalServerError, STATUS_ERR_FINDING_USR, err.Error())
+					sendModelAsResWithStatus(res, status.NewStatus(http.StatusInternalServerError, STATUS_ERR_FINDING_USR), http.StatusInternalServerError)
+					return
+				} else if len(results) > 0 {
+					a.logger.Println(http.StatusConflict, STATUS_USR_ALREADY_EXISTS)
+					sendModelAsResWithStatus(res, status.NewStatus(http.StatusConflict, STATUS_USR_ALREADY_EXISTS), http.StatusConflict)
+					return
 				}
+			}
+			//Rehash the pw if needed
+			if updatesToApply.Updates.Pw != "" {
 
-				//All good - now update
-				if err := a.Store.UpsertUser(userToUpdate); err != nil {
-					log.Printf(USER_API_PREFIX+"UpdateUser %s err[%s]", STATUS_ERR_UPDATING_USR, err.Error())
+				if err := userToUpdate.HashPassword(updatesToApply.Updates.Pw, a.ApiConfig.Salt); err != nil {
+					a.logger.Println(http.StatusInternalServerError, STATUS_ERR_UPDATING_USR, err.Error())
 					sendModelAsResWithStatus(res, status.NewStatus(http.StatusInternalServerError, STATUS_ERR_UPDATING_USR), http.StatusInternalServerError)
 					return
+				}
+			}
+
+			//All good - now update
+			if err := a.Store.UpsertUser(userToUpdate); err != nil {
+				a.logger.Println(http.StatusInternalServerError, STATUS_ERR_UPDATING_USR, err.Error())
+				sendModelAsResWithStatus(res, status.NewStatus(http.StatusInternalServerError, STATUS_ERR_UPDATING_USR), http.StatusInternalServerError)
+				return
+			} else {
+				if td.IsServer {
+					a.logMetricForUser(usrId, "userupdated", req.Header.Get(TP_SESSION_TOKEN), map[string]string{"server": "true"})
 				} else {
-					if td.IsServer {
-						a.logMetricForUser(usrId, "userupdated", req.Header.Get(TP_SESSION_TOKEN), map[string]string{"server": "true"})
-					} else {
-						a.logMetric("userupdated", req.Header.Get(TP_SESSION_TOKEN), map[string]string{"server": "false"})
-					}
-					res.WriteHeader(http.StatusOK)
-					return
+					a.logMetric("userupdated", req.Header.Get(TP_SESSION_TOKEN), map[string]string{"server": "false"})
 				}
+				res.WriteHeader(http.StatusOK)
+				return
 			}
 		}
 	}
-	res.WriteHeader(http.StatusUnauthorized)
```

Comment:
We do this as part of the token check now - if there is an issue with the token or its not valid then that happens first thing - the actual error is logged also